### PR TITLE
Move Engine API to Powchain and Consolidate Connection Management

### LINKED ISSUE #10465: Panic: event: wrong type in Subscribe got chan
I got panic when starting the beacon node. This will happen if config `BellatrixForkEpoch` has been set to non-max. Looks related to #10413, we are passing in wrong type of channel

```
panic: event: wrong type in Subscribe got chan *state.BlockProcessedData, want chan<- *feed.Event

goroutine 227 [running]:
github.com/prysmaticlabs/prysm/async/event.(*Feed).Subscribe(0xc000ddfc00, {0x528a0c0, 0xc002a2e660})
	async/event/feed.go:86 +0x4ea
github.com/prysmaticlabs/prysm/beacon-chain/powchain.(*Service).checkTransitionConfiguration(0xc000030c40, {0x5aa1af0, 0xc0065f4880}, 0xc002a2e660)
	beacon-chain/powchain/check_transition_config.go:62 +0x616
created by github.com/prysmaticlabs/prysm/beacon-chain/powchain.(*Service).Start
	beacon-chain/powchain/service.go:255 +0x30f
```

### PR DESCRIPTION
**Pull request by rauljordan**
## What does this PR do? Why is it needed?

Closes #10465 

Three things:
1. Move the engine API client abstraction into becoming part of the `powchain` service
2. Consolidate connection management for Prysm to connect to execution nodes in a single place
3. Connect to execution nodes on initialization rather than at service runtime

### Moving the Engine API Client

Previously, we had created an `Engine API Client` abstraction in its own package in Prysm. However, it makes more sense to move this as part of the `powchain` service, as that service already manages an execution client's json-rpc connection. This simplifies our logic and makes things far more concise.

### Consolidating Connection Management for ETH RPC Endpoints 

Ethereum execution clients, such as go-ethereum, expose a json-rpc API on localhost:8545 by default, offering methods in the `eth` RPC namespace. For the Bellatrix hard-fork, Prysm will require a connection to an execution node's `engine` json-rpc API namespace. Originally, go-ethereum had planned to expose this in a totally different port than the normal RPC endpoint. However, this was scrapped and now it is available at localhost:8545 albeit with authentication as a requirement.

### Connecting to ETH RPC on Initialization

For the merge, we need to connect to RPC endpoints on initialization rather than in runtime Start() functions, as we do today. This is a fairly invasive change, and here is how we deal with it:

1. Move the dialing of ETH endpoints to the powchain's `New()` function
2. Ensure we can dial RPC endpoints even if the ETH client is not synced. Previously, we would have complex connection management logic to check sync status of the execution client and disconnect if it fell out of sync
3. Ensure tests are updated to pass these requirements

### CODE REVIEW

**Comment by prestonvanloon** [comment]
Some build failures.
```
cmd/beacon-chain/powchain/options.go:30:23: undefined: powchain.WithExecutionClientJWTSecret
```

File: beacon-chain/powchain/service.go [hunk 1]
```diff
@@ -212,6 +213,14 @@ func NewService(ctx context.Context, opts ...Option) (*Service, error) {
 		}
 	}
 
+	if err := s.connectToPowChain(); err != nil {
```
Comment (rauljordan) [inline beacon-chain/powchain/service.go:216]: We now connect to the powchain on initialization, as this is required for the merge

File: beacon-chain/powchain/service.go [hunk 2]
```diff
@@ -424,15 +421,6 @@ func (s *Service) dialETH1Nodes(endpoint network.Endpoint) (*ethclient.Client, *
...
 	}
-	syncProg, err := httpClient.SyncProgress(s.ctx)
-	if err != nil {
-		closeClients()
-		return nil, nil, err
-	}
-	if syncProg != nil {
-		closeClients()
-		return nil, nil, errors.New("eth1 node has not finished syncing yet")
-	}
```
Comment (rauljordan) [inline beacon-chain/powchain/service.go:435]: We would avoid connecting to geth if it is syncing. However, this behavior needs to change for the merge. We must connect to geth even if it is syncing

File: beacon-chain/powchain/service.go [hunk 3]
```diff
@@ -456,17 +444,6 @@ func (s *Service) dialETH1Nodes(endpoint network.Endpoint) (*ethclient.Client, *
 	return httpClient, httpRPCClient, nil
 }
 
-func (s *Service) initializeConnection(
```
Comment (rauljordan) [inline beacon-chain/powchain/service.go:459]: Not necessary to have this as its own func

File: beacon-chain/powchain/service.go [hunk 4]
```diff
@@ -479,30 +456,8 @@ func (s *Service) closeClients() {
 	}
 }
 
-func (s *Service) waitForConnection() {
```
Comment (rauljordan) [inline beacon-chain/powchain/service.go:482]: This function had some complex logic around checking for eth1 nodes falling out of sync. However, we cannot use this behavior for the merge

File: beacon-chain/powchain/service.go [hunk 5]
```diff
@@ -1056,19 +996,6 @@ func (s *Service) ensureValidPowchainData(ctx context.Context) error {
 	return nil
 }
 
-// Initializes a connection to the engine API if an execution provider endpoint is set.
-func (s *Service) initializeEngineAPIClient(ctx context.Context) error {
```
Comment (rauljordan) [inline beacon-chain/powchain/service.go:1060]: this is no longer a separate entity from the normal RPC connection

**Review by terencechain — APPROVED** [review]
This PR looks good to me from the technical perspective. I also tested the PR with and without `--execution_provider` flag on the `kiln` testnet

File: beacon-chain/powchain/options.go [hunk 6]
```diff
@@ -10,6 +13,9 @@ import (
 	"github.com/prysmaticlabs/prysm/network"
 )
 
+// DefaultRPCHTTPTimeout for HTTP requests via an RPC connection to an execution node.
+const DefaultRPCHTTPTimeout = time.Second * 5
```
Comment (nisdas) [inline beacon-chain/powchain/options.go:17]: we need to be careful with this number, some our requests when syncing deposit logs require the execution node to answer with large payloads. These can take some time.

Comment (rauljordan) [reply beacon-chain/powchain/options.go:17]: half a slot should be sufficient, perhaps. What do you think about 6 seconds @terencechain ?

Comment (terencechain) [reply beacon-chain/powchain/options.go]: Six is fine with me. Ideally, the different RPC calls should have different timeout constraints but we could add that later

**Pull request merged**